KerbalStuff/config.py

```diff
@@ -15,13 +15,27 @@ def get_env_var_or_config(section, key):
     if env_var:
         return env_var
     else:
-        return config.get(section, key)
+        return config.get(section, key, fallback=None)
 
 
-_cfg = lambda k: get_env_var_or_config(env, k)
-_cfgi = lambda k: int(_cfg(k))
-_cfgb = lambda k: strtobool(_cfg(k)) == 1
-_cfgl = lambda k: ast.literal_eval(_cfg(k))
+def _cfg(k):
+    return get_env_var_or_config(env, k)
+
+
+def _cfgi(k, default = 0):
+    val = _cfg(k)
+    return int(val) if val is not None else default
+
+
+def _cfgb(k, default = False):
+    val = _cfg(k)
+    return strtobool(val) == 1 if val is not None else default
+
+
+def _cfgl(k, default = {}):
```

Comment:
There's one other reason why you can't make `{}` as `default =` argument.
The `{}` is a `dict` object which is saved inside the `_cfgl` object and passed not as is to _all_ calls.
So in this case all calls to `_cfgl` that couldn't find the value in the config would return the exact same object, not a new one each time.
```suggestion
def _cfgl(k, default = None):
```

Reply:
That's fine, because app code will not be setting these values. Values will only come from the `config.ini` file.

Reply:
That's _really, really not fine._ It's one of _THE_ worst junior mistakes in Python. You can't make assumptions as to what the code will do with the result of your function, because this breaks incapsulation. And you never ever use mutable objects as default arguments.

Reply:
OK, switched the parameter default to `None` and set it to `{}` inside the function when it's `None`.